### PR TITLE
[Test] Increase registered callback's duration to avoid flakyness

### PR DESCRIPTION
**Pull request by yijiem**
Previously the registered callback's duration is set too low (10ms) that 2 different OTel callbacks get called from OTel consecutively would trigger the registered callback again, making the test flaky.

### CODE REVIEW

File: test/cpp/ext/otel/otel_plugin_test.cc [hunk 1]
```diff
@@ -1956,8 +1956,8 @@ TEST_F(OpenTelemetryPluginCallbackMetricsTest,
       data = collector.Stop();
   // Verify that data is incremental without duplications (cached
   // values).
-  EXPECT_EQ(report_count_1, kIterations);
-  EXPECT_EQ(report_count_2, kIterations);
+  EXPECT_GE(report_count_1, kIterations);
```
Comment (yashykt) [inline test/cpp/ext/otel/otel_plugin_test.cc:1959]: can you dig into why this might be happening?

Comment (yijiem) [reply test/cpp/ext/otel/otel_plugin_test.cc]: reverted back and change the duration instead

**Review by yashykt — APPROVED** [review]
Thanks!

File: test/cpp/ext/otel/otel_plugin_test.cc [hunk 2]
```diff
@@ -1936,9 +1936,9 @@ TEST_F(OpenTelemetryPluginCallbackMetricsTest,
         reporter.Report(double_gauge_handle, double_value_2++, kLabelValuesSet2,
                         kOptionalLabelValuesSet2);
       },
-      grpc_core::Duration::Milliseconds(10) * grpc_test_slowdown_factor(),
+      grpc_core::Duration::Milliseconds(50) * grpc_test_slowdown_factor(),
       integer_gauge_handle, double_gauge_handle);
-  constexpr int kIterations = 100;
+  constexpr int kIterations = 50;
```
Comment (yashykt) [inline test/cpp/ext/otel/otel_plugin_test.cc:1941]: maybe reduce it to 20?

**Pull request closed without merging**